metis-dereference/metis-dereference-import/src/main/java/eu/europeana/metis/dereference/vocimport/VocabularyCollectionImporterFactory.java

```diff
@@ -12,25 +15,44 @@
  */
 public class VocabularyCollectionImporterFactory {
 
+  private final DereferenceValidationUtils dereferenceValidationUtils;
+
+  /**
+   * Constructor for the factory
+   *
+   * @param dereferenceValidationUtils The utils class used to verify input values
+   */
+  public VocabularyCollectionImporterFactory(DereferenceValidationUtils dereferenceValidationUtils) {
+    this.dereferenceValidationUtils = dereferenceValidationUtils;
+  }
+
   /**
    * Create a vocabulary importer for remote web addresses, indicated by instances of {@link URI}.
    * Note that this method can only be used for locations that are also a valid {@link
    * java.net.URL}.
    *
    * @param directoryLocation The location of the directory to import.
+   * @throws VocabularyImportException if a problem occurs when verifying directory
    * @return A vocabulary importer.
    */
-  public VocabularyCollectionImporter createImporter(URI directoryLocation) {
+  public VocabularyCollectionImporter createImporter(URI directoryLocation) throws VocabularyImportException {
+    if(!dereferenceValidationUtils.isDirectoryValid(directoryLocation.toString())){
+      throw new VocabularyImportException("The location of the directory to import is not valid.");
+    }
     return new VocabularyCollectionImporterImpl(new UriLocation(directoryLocation));
   }
 
   /**
    * Create a vocabulary importer for local files, indicated by instances of {@link Path}.
    *
    * @param directoryLocation The location of the directory file to import.
+   * @throws VocabularyImportException if a problem occurs when verifying directory
    * @return A vocabulary importer.
    */
-  public VocabularyCollectionImporter createImporter(Path directoryLocation) {
+  public VocabularyCollectionImporter createImporter(Path directoryLocation) throws VocabularyImportException {
+    if(!dereferenceValidationUtils.isDirectoryValid(directoryLocation.toString())){
```

Comment:
How about inverting this?
So check for true instead of "!" and return in the if, and outside of if throw.
Makes it slightly more readable.
(Same for the other occasion)

Reply:
I made it has it is so it would be simple for me to follow. But that is a good point, since it is always used to check the negative of it. I will change this